core/src/main/scala/cilib/Boundary.scala

```diff
@@ -0,0 +1,82 @@
+package cilib
+
+import scalaz._
+import Scalaz._
+import spire.math.Interval
+
+object Boundary {
+
+  final case class Enforce[F[_], A](f: (A, Interval[Double]) => F[A])
+  final case class EnforceTo[F[_], A](f: (A, Interval[Double], A) => F[A])
+
+  def enforce[F[_]: Applicative, A: spire.math.Numeric](x: Position[A],
+                                                        f: Enforce[F, A]): F[NonEmptyList[A]] =
+    x.pos.zip(x.boundary).traverse { case (a, b) => f.f(a, b) }
+
+  def enforceTo[F[_], A: spire.math.Numeric, B](x: Position[A], z: Position[A], f: EnforceTo[F, A])(
+      implicit F: Applicative[F]): F[NonEmptyList[A]] =
+    x.pos.zip(x.boundary).zip(z.pos).traverse {
+      case ((a, b), c) => f.f(a, b, c)
+    }
+
+  def clamp[A](implicit N: spire.math.Numeric[A]) =
+    absorb
+
+  def absorb[A](implicit N: spire.math.Numeric[A]) =
+    Enforce((a: A, b: Interval[Double]) =>
+      Need {
+        val z = N.toDouble(a)
+
+        if (z < b.lowerValue) N.fromDouble(b.lowerValue)
+        else if (z > b.upperValue) N.fromDouble(b.upperValue)
+        else a
+    })
+
+  def random[A](implicit N: spire.math.Numeric[A]) =
+    Enforce(
+      (a: A, b: Interval[Double]) =>
+        if (b.contains(N.toDouble(a))) RVar.pure(a)
+        else Dist.uniform(b).map(N.fromDouble))
+
+  def reflect[A](implicit N: spire.math.Numeric[A]) =
+    Enforce((a: A, b: Interval[Double]) =>
+      Need {
+        val z = N.toDouble(a)
+
+        if (z < b.lowerValue) N.fromDouble(b.lowerValue - (z - b.lowerValue))
+        else if (z > b.upperValue) N.fromDouble(b.upperValue - (z - b.upperValue))
```

Comment:
What about cases where the reflection makes it go passed the other bound? e.g. ```b=(5,6) a=3``` the reflected value is 7

Reply:
Looking at some more literature, the paper I originally was basing these equations off was actually wrong. The equation has been adjusted and some tests have been added to ensure that they are correctly within the bounds.